server/src/controllers/gmail.js

```diff
@@ -0,0 +1,80 @@
+const { google } = require("googleapis");
+const DataBaseConnectionError = require("../errors/database-connection-error");
+const db = require("../../database/models/index");
+
+// The controller for Gmail Client
+class Gmail {
+  // Get an OAuthClient with configurations
+  static get_oAuthClient = () => {
+    const client_id = process.env.CLIENT_ID;
+    const client_secret = process.env.CLIENT_SECRET;
+    const redirect_uris = "http://localhost:3000/gmailauth";
```

Comment:
Should probably pull these vairalbes from a config file so they are more easy to edit.. especially since we will have to edit redirect before deployment